keps/sig-node/5359-pod-level-swap-control/README.md

```diff
@@ -0,0 +1,1115 @@
+their workload's intolerance to swap.
+
+This KEP addresses this gap by providing a pod-level swap control API.
+In the first phase, we'll focus on the ability to disable swap for all its containers irrespective of underlying node
+swap behavior.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+* Introduce a new `swapPolicy` field in PodSpec to allow users to configure swap for an individual pod.
+* Initially, provide a way to disable swap under the `LimitedSwap` swap behavior.
+* Maintain backward compatibility: existing pods that run on swap-enabled nodes should behave as they do by default.
+* Provide a mechanism to alleviate concerns regarding "all-or-nothing" nature of node-level swap enablement, potentially
+  unblocking KEP-2400’s path to GA.
+* Open the door to future enhancements, both at the kubelet-level swap behaviors and the new pod-level swap control API
+  field.
+  For example, a more sophisticated declarative node-level swap behavior that relies on hints provided by pod owners,
+  or a swap behavior that allows users to explicitly specify swap limits for individual pods.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* To introduce fine-grained swap controls at the pod or container level (e.g., setting a specific swap limit other than
+  zero) in this initial KEP.
+* To change the fundamental behavior of LimitedSwap for Guaranteed, Burstable, or BestEffort pods when the pod-level
+  swap setting is not specified.
+* To define how swap contributes to pod eviction or schedulable resource limits beyond disabling its usage.
+* To enhance pod scheduling to allow users to specify swap-capable node preferences. This will be achievable with
+  features such
+  as [node-capabilities](https://docs.google.com/document/d/1vSDlAA3o0riVq0EcmGBOYUJUVF4tN2Ib7VJg3o1LvBw/edit?tab=t.0).
+* To support cgroupv1 for this feature, as Kubernetes swap support (KEP 2400) is focused on cgroup v2.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As an application owner of a latency-critical service, I want to explicitly ensure my pods never use swap in my
```

Comment:
I have said elsewhere but I feel obligated to repeat - this user-story IS NOT satisfied by the KEP, as I understand it.

A guaranteed-QoS pod which has disabled swap /for itself/ might still get stuck waiting for swap-out /in another pod/, in order to reclaim pages.

Reply:
You're obviously right.

I guess it's a matter of terminology and how you understand `ensure my pods never use swap`.
I've tried to phrase it better now. WDYT?